client/js/components/statement/listStatements/ExportModal.vue

```diff
@@ -0,0 +1,241 @@
+          dataCy: 'exportModal:input:colRight',
+          placeholder: Translator.trans('segment.recommendation'),
+          title: null
+        }
+      },
+      exportTypes: {
+        docx: {
+          label: 'export.statements.docx',
+          uploadPath: 'dplan_statement_segments_export',
+          dataCy: 'exportModal:export:docx'
+        },
+        zip: {
+          label: 'export.statements.zip',
+          uploadPath: 'dplan_statement_segments_export_packaged',
+          dataCy: 'exportModal:export:zip'
+        },
+        xlsx: {
+          label: 'export.statements.xlsx',
+          uploadPath: 'dplan_statement_xls_export',
+          dataCy: 'exportModal:export:xlsx'
+        }
+      }
+    }
+  },
+
+  methods: {
+    applyDefaultPlaceholdersForEmptyNames () {
+      Object.keys(this.docxColumns).forEach(key => {
+        if (!this.docxColumns[key].title) {
+          this.docxColumns[key].title = null
+          this.docxColumns[key].placeholder = this.getDefaultPlaceholderByKey(key)
+        }
+      })
+    },
+
+    closeModalAndResetValues () {
+      this.$refs.exportModalInner.toggle()
+      this.active = 'docx'
+      this.resetColumnsUnsavedValues()
+    },
+
+    getDefaultPlaceholderByKey (key) {
+      const defaultPlaceholders = {
+        left: Translator.trans('segments.export.segment.id'),
+        middle: Translator.trans('segments.export.statement.label'),
+        right: Translator.trans('segment.recommendation')
+      }
+
+      return defaultPlaceholders[key]
+    },
+
+    handleAbort () {
+      this.$emit('abort')
+      this.closeModalAndResetValues()
+    },
+
+    handleExport () {
+      const columnTitles = {}
+
+      Object.keys(this.docxColumns).forEach(key => {
+        const columnTitle = this.docxColumns[key].title
+        const storageKey = `exportModal:docxCol:${key}`
+
+        if (columnTitle) {
+          this.updateSessionStorage(storageKey, columnTitle)
+          this.docxColumns[key].placeholder = columnTitle
+          columnTitles[key] = columnTitle
+        } else {
+          this.removeFromSessionStorage(storageKey)
+          this.docxColumns[key].placeholder = this.getDefaultPlaceholderByKey(key)
+          columnTitles[key] = null /** Setting the value to null will trigger the display of the default column titles */
+        }
+      })
+
+      this.$emit('export', {
+        route: this.exportTypes[this.active].uploadPath,
+        docxHeaders: ['docx', 'zip'].includes(this.active) ? columnTitles : null
+      })
+      this.closeModalAndResetValues()
+    },
+
+    handleInput () {
+      this.applyDefaultPlaceholdersForEmptyNames()
+    },
+
+    isActive (key) {
+      return key === this.active
+    },
+
+    openModal () {
+      this.setInitialValues()
+      this.$refs.exportModalInner.toggle()
+    },
+
+    resetColumnsUnsavedValues () {
+      Object.keys(this.docxColumns).forEach(key => {
+        this.setTitleAndPlaceholderByKey(key)
+      })
+    },
```

Comment:
this is redundant with setInitialValues. I think we can delete this one here and only set the values on open. Doing it twice is a bit unnecessary ;-)